src/coreclr/vm/stringliteralmap.cpp

```diff
@@ -489,7 +489,7 @@ StringLiteralEntry *GlobalStringLiteralMap::AddStringLiteral(EEStringData *pStri
     {
     PinnedHeapHandleBlockHolder pStrObj(&m_PinnedHeapHandleTable,1);
     // Create the COM+ string object.
-    STRINGREF strObj = AllocateStringObject(pStringData);
+    STRINGREF strObj = AllocateStringObject(pStringData, true);
```

Comment:
This will need changes on the `AddInternedString` path as well to guarantee that all strings in this table are pinned.